### ISSUE TITLE
Syncthing-Fork crashes during startup if Syncthing-Android is already running

### ISSUE THREAD

**Issue by o-l-a-v** [Catfriend1/syncthing-android#193]
If you have issues with this fork, feel free to post them. I'll look into them if a good reproducer or log is provided.

### Description of the issue
Syncthing-Fork crashes during startup if Syncthing-Android is already running.
No error messages is given, Syncthing-Fork only force closes.

### Reproducer
* Install both Syncthing-Android and Syncthing-Fork on the same system.
* Start Syncthing-Android
* Start Syncthing-Fork

### Version Information
    App Version: 1.0.0.1
    Syncthing Version: 1.0.0.0
    Android Version: Android 8.0.0
    System: nVidia Shield Android TV 2015 Pro, Shield Experience Update v7.2.1

### Android Log
    adb shell / logcat
    or MatLog (search on GitHub) output
[logs-2019-01-04-20-06-36.zip](https://github.com/Catfriend1/syncthing-android/files/2728193/logs-2019-01-04-20-06-36.zip)

**Comment by Catfriend1** [comment]
Yes, I already know about that. You cannot run both are same time because the native binary has the same name internally. Maybe I could detect this and display an error message if it doesn't hang startup forever.

**Comment by o-l-a-v** [comment]
A error message would be nice. 
I understand that they both can't coexist. Uninstalled the Syncthing-Android app, only running yours now. Thanks!

**Comment by Catfriend1** [comment]
Well, you've got:
```
SyncthingService(12594): Starting syncthing
onServiceStateChange: from DISABLED to STARTING
Web GUI will be available at https://127.0.0.1:45335 << because the default port 8384 isn't available because of the other instance > this is bad as startup cannot be detected as "finished".
runShellCommand: kill -SIGINT 11682 << Trying to kill the syncthing process also fails, as the native process is owned by "the other app" from android's perspective thus failing with exit code 1.
```
We're then waiting "forever" that SyncthingNative exists which will never happen because of the failed kill command >> this triggers android's "crash handler", the app is just not responding (see ANR line) but not "crashed", android marks it as crashed and terminates it forcefully.

This is hard to cleanup as both native procs are called "libSyncthing.so" and our instance sees both. Logically Android "should" be like the "kill", so we just see processes belonging to our app and not a "foreign app" running a process with the same name. Again, Android is a bit beasty here. I'll come back if I can imagine a proper solution. I don't want to catch the kill exit code 1 as I'm not sure if it always refers to this problem so probably catching it would confuse users.

At all reading this: If anyone knows how to detect the second instance from another app properly, filing a PR would be of great help here :)
